src/features/filter.ts

```diff
@@ -1,10 +1,64 @@
-export const actions = { /* put action creators here */};
+type Filter = {
+  query: string,
+  status: string,
+};
+
+type FilterStatus = {
+  type: 'filter/STATUS';
+  query: string;
+  status: string,
+};
+type FilterQuery = {
+  type: 'filter/QUERY';
+  query: string;
+  status: string,
+};
+type FilterClear = {
+  type: 'filter/CLEAR';
+  query: string;
+  status: string,
+};
+
+type Action = FilterStatus
+| FilterQuery
+| FilterClear;
+
+const filterStatus = (status: string, query = ''): FilterStatus => ({
+  type: 'filter/STATUS',
+  query,
```

Comment:
as a common practice action should contain two fields - type and payload. The same as you did for setTodos action